src/WinRT.Runtime2/InteropServices/RestrictedErrorInfo.cs

```diff
@@ -62,26 +204,125 @@ public static void ThrowExceptionForHR(HRESULT errorCode)
     /// <seealso cref="Marshal.GetExceptionForHR(int)"/>
     public static HRESULT GetHRForException(Exception? exception)
     {
-        // TODO
-        return 0;
+        ArgumentNullException.ThrowIfNull(exception);
+        int hresult = exception.HResult;
```

Comment:
Always use `HRESULT` for hresults (it's an alias):
```suggestion
        HRESULT hresult = exception.HResult;
```